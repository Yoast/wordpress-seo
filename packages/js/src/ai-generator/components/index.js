export { App } from "./app";
export { GoogleContent } from "./google-content";
export { GooglePreview } from "./google-preview";
export { GooglePreviewSkeleton } from "./google-preview-skeleton";
export { Introduction } from "./introduction";
export { LengthProgressBar } from "./length-progress-bar";
export { ModalContent } from "./modal-content";
export { PreviewModePicker } from "./preview-mode-picker";
export { SocialContent } from "./social-content";
export { SocialPreview } from "./social-preview";
export { SocialPreviewSkeleton } from "./social-preview-skeleton";
export { SuggestionError } from "./suggestion-error";
export { SuggestionsList } from "./suggestions-list";
export { SuggestionsListSkeleton } from "./suggestions-list-skeleton";
export { TipNotification } from "./tip-notification";
export { TwitterContent } from "./twitter-content";
export { TwitterPreview } from "./twitter-preview";
export { TwitterPreviewSkeleton } from "./twitter-preview-skeleton";
export { TypeProvider } from "./type-provider";
export { SparksLimitNotification } from "./sparks-limit-notification";
