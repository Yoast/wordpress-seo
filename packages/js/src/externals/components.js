import CollapsibleCornerstone from "../components/CollapsibleCornerstone";
import KeywordInput from "../components/contentAnalysis/KeywordInput";
import ReadabilityAnalysis from "../components/contentAnalysis/ReadabilityAnalysis";
import SeoAnalysis from "../components/contentAnalysis/SeoAnalysis";

window.yoast = window.yoast || {};
window.yoast.externals = window.yoast.externals || {};
window.yoast.externals.components = {
	CollapsibleCornerstone,
	KeywordInput,
	ReadabilityAnalysis,
	SeoAnalysis,
};
