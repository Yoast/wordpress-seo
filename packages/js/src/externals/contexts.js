import { LocationConsumer, LocationContext, LocationProvider } from "../components/contexts/location";

window.yoast = window.yoast || {};
window.yoast.externals = window.yoast.externals || {};
window.yoast.externals.contexts = {
	LocationContext,
	LocationProvider,
	LocationConsumer,
};
