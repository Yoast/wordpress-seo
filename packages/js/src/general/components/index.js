export { AlertsList } from "./alerts-list";
export { AlertsTitle } from "./alerts-title";
export { Collapsible } from "./collapsible";
export { Notice } from "./notice";
export { Notifications } from "./notifications";
export { Problems } from "./problems";
export { RouteErrorFallback } from "./route-error-fallback";
export { RouteLayout } from "./route-layout";
