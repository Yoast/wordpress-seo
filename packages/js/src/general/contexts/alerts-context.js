import { createContext } from "@wordpress/element";

/**
 * The context for the alerts.
 */
export const AlertsContext = createContext( { Icon: null, bulletClass: "", iconClass: "" } );
