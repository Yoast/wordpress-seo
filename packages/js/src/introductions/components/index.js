export { Content } from "./content";
export { Introduction } from "./introduction";
export { Modal } from "./modal";
export { IntroductionProvider, useIntroductionsContext } from "./provider";
