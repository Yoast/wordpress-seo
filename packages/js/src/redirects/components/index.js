export { RouteErrorFallback } from "./route-error-fallback";
export { RouteLayout } from "./route-layout";
export { Menu } from "./menu";
export { SidebarLayout } from "./sidebar-layout";
export { Notifications } from "./notifications";
