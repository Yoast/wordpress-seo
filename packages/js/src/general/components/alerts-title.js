import { useContext } from "@wordpress/element";
import PropTypes from "prop-types";
import classNames from "classnames";
import { ExclamationCircleIcon } from "@heroicons/react/outline";
import { Title } from "@yoast/ui-library";
import { AlertsContext } from "../contexts/alerts-context";

/**
 *
 * @param {string} title The title of alerts.
 * @param {number} counts The count of the alerts.
 * @param {JSX.Node} children The children.
 *
 * @returns {JSX.Element} The alert title element.
 */
export const AlertsTitle = ( {
	title,
	counts = 0,
	children = null,
} ) => {
	const { Icon = ExclamationCircleIcon, iconClass = "" } = useContext( AlertsContext );

	return (
		<div>
			<div className="yst-flex yst-justify-between yst-gap-2 yst-items-center">
				<Icon className={ classNames( "yst-w-6 yst-h-6", iconClass ) } />
				<Title className="yst-grow" as="h2" size="2">{ title } { `(${ counts })` }</Title>
			</div>
			{ children }
		</div>
	);
};

AlertsTitle.propTypes = {
	title: PropTypes.string,
	counts: PropTypes.number,
	children: PropTypes.node,
};
