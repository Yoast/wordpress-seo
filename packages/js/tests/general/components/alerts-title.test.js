import { render } from "../../test-utils";
import { AlertsTitle } from "../../../src/general/components/alerts-title";
import { AlertsContext } from "../../../src/general/contexts/alerts-context";
import { BellIcon } from "@heroicons/react/outline";

const notificationsTheme = {
	Icon: BellIcon,
	bulletClass: "yst-fill-blue-500",
	iconClass: "yst-text-blue-500",
};

describe( "AlertsTitle", () => {
	it( "should match snapshot", () => {
		const title = "Test Title";
		const counts = 5;
		const { container } = render( <AlertsContext.Provider value={ notificationsTheme }>
			<AlertsTitle title={ title } counts={ counts } />
		</AlertsContext.Provider> );
		expect( container ).toMatchSnapshot();
	} );

	it( "renders the children correctly", () => {
		const children = <div>Test Children</div>;

		const { getByText } = render(
			<AlertsContext.Provider value={ notificationsTheme }>
				<AlertsTitle title="Test Title" counts={ 0 }>
					{ children }
				</AlertsTitle>
			</AlertsContext.Provider>
		);

		const childrenElement = getByText( "Test Children" );
		expect( childrenElement ).toBeInTheDocument();
	} );
} );
