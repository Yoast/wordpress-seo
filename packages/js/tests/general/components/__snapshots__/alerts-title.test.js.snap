// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertsTitle should match snapshot 1`] = `
<div>
  <div>
    <div
      class="yst-flex yst-justify-between yst-gap-2 yst-items-center"
    >
      <svg
        aria-hidden="true"
        class="yst-w-6 yst-h-6 yst-text-blue-500"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 17h5l-1.405-1.405A2.032 2.032 0 0118 14.158V11a6.002 6.002 0 00-4-5.659V5a2 2 0 10-4 0v.341C7.67 6.165 6 8.388 6 11v3.159c0 .538-.214 1.055-.595 1.436L4 17h5m6 0v1a3 3 0 11-6 0v-1m6 0H9"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <h2
        class="yst-title yst-title--2 yst-grow"
      >
        Test Title
         
        (5)
      </h2>
    </div>
  </div>
</div>
`;
